Tasks/Common/azure-arm-rest/azure-app-service.ts

```diff
@@ -0,0 +1,185 @@
+        httpRequest.method = 'POST';
+        httpRequest.uri = this.azureServiceClient.getRequestUri('//subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/sites/{AppServiceName}/{slotUrl}/config/appsettings/list',
+            {
+                '{ResourceGroupName}': resourceGroupName,
+                '{AppServiceName}': this.appServiceName,
+                '{slotUrl}': slotUrl
+            }, null, '2016-08-01');
+
+        this.azureServiceClient.beginRequest(httpRequest).then((response: webClient.WebResponse) => {
+            if(response.statusCode == 200) {
+                dataDeferred.resolve(response.body);
+            }
+            else {
+                dataDeferred.reject("Failed for get App Details: " + JSON.stringify(response));    
+            }
+        }, (error) => {
+            dataDeferred.reject("Failed for get App Details: " + error.toString())
+        });
+
+        return dataDeferred.promise;
+    }
+
+    public async getAppDetails(force?: boolean) {
+        var dataDeferred = Q.defer<any>();
+        if(this.appDetails && !force) {
+            console.log("getAppDetails in cache");
+            dataDeferred.resolve(this.appDetails);
+        }
+        else {
+            console.log(" getAppDetails not in cache");
+            var resourceGroupName: string = await this.getResourceGroupName();
+            var slotUrl: string = this.slotName ? `/slots/${this.slotName}` : '';
+            var httpRequest = new webClient.WebRequest();
+            httpRequest.method = 'GET';
+            httpRequest.uri = this.azureServiceClient.getRequestUri('//subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/sites/{AppServiceName}/{slotUrl}',
+                {
+                    '{ResourceGroupName}': resourceGroupName,
+                    '{AppServiceName}': this.appServiceName,
+                    '{slotUrl}': slotUrl
+                }, null, '2016-08-01');
+
+            this.azureServiceClient.beginRequest(httpRequest).then((response: webClient.WebResponse) => {
+                if(response.statusCode == 200) {
+                    this.appDetails = response.body;
+                    dataDeferred.resolve(this.appDetails);
+                }
+                else {
+
+                    dataDeferred.reject("Failed for get App Details: " + JSON.stringify(response));    
+                }
+            }, (error) => {
+                dataDeferred.reject("Failed for get App Details: " + error.toString())
+            });
+        }
+
+        return dataDeferred.promise;
+    }
+
+    // Move to azure-arm-common - Azure Monitor tasks
+    private async getResourceID() {
+        var dataDeferred = Q.defer<any>();
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'GET';
+        httpRequest.uri = this.azureServiceClient.getRequestUri('//subscriptions/{subscriptionId}/resources', null,
```

Comment:
two slashes before subscriptions in uri ?

Reply:
following the same pattern used by RG team. Anyway, the forward slashes are removed in the getRequestUri function